diff --git a/usr/share/lipstick-jolla-home-qt5/switcher/SwitcherItem.qml b/usr/share/lipstick-jolla-home-qt5/switcher/SwitcherItem.qml
index b6c281a..afd7208 100644
--- a/usr/share/lipstick-jolla-home-qt5/switcher/SwitcherItem.qml
+++ b/usr/share/lipstick-jolla-home-qt5/switcher/SwitcherItem.qml
@@ -327,7 +327,10 @@ EditableGridDelegate {
                             Behavior on opacity { FadeAnimation {} }
                         }
 
-                        onClicked: coverActionModel.trigger(model.id)
+                        onClicked: {
+                            coverActionModel.trigger(model.id)
+                            switcherRoot.minimizeLaunchingWindows()
+                        }
                     }
                 }
             }
